coderd/database/queries/oauth2.sql

```diff
@@ -129,6 +129,11 @@ SELECT * FROM oauth2_provider_app_codes WHERE id = $1;
 -- name: GetOAuth2ProviderAppCodeByPrefix :one
 SELECT * FROM oauth2_provider_app_codes WHERE secret_prefix = $1;
 
+-- name: ConsumeOAuth2ProviderAppCodeByPrefix :one
```

Comment:
Since you're only expecting one row back, you should add a `LIMIT 1` just to be safe.

Reply:
So there is no `LIMIT 1` for DELETE FROM's or UPDATEs.

I could rewrite that query to the following according to postgres docs (https://www.postgresql.org/docs/current/sql-delete.html):

```
While there is no LIMIT clause for DELETE, it is possible to get a similar effect using the same method described in the documentation of UPDATE:

WITH delete_batch AS (
  SELECT l.ctid FROM user_logs AS l
    WHERE l.status = 'archived'
    ORDER BY l.creation_date
    FOR UPDATE
    LIMIT 10000
)
DELETE FROM user_logs AS dl
  USING delete_batch AS del
  WHERE dl.ctid = del.ctid;
```

Reply:
I'm not entirely sure if that added complexity outweighs the "safety" of limiting it to 1.